Multiple files: reduce padding in structures

mykaul opened the pull request:
Examples:
struct gf_log_handle_
        /* size: 208, cachelines: 4, members: 22 */
        /* sum members: 194, holes: 2, sum holes: 8 */
        /* padding: 6 */
        /* last cacheline: 16 bytes */
After:
        /* size: 200, cachelines: 4, members: 22 */
        /* padding: 6 */
        /* last cacheline: 8 bytes */

struct clienttable {
        /* size: 72, cachelines: 2, members: 5 */
        /* sum members: 64, holes: 2, sum holes: 8 */
        /* last cacheline: 8 bytes */
After:
        /* size: 64, cachelines: 1, members: 5 */

struct gf_flock {
       /* size: 1056, cachelines: 17, members: 6 */
        /* sum members: 1052, holes: 1, sum holes: 4 */
        /* last cacheline: 32 bytes */
After:
        /* size: 1056, cachelines: 17, members: 6 */
        /* padding: 4 */
        /* last cacheline: 32 bytes */

struct _call_stack
        /* size: 1864, cachelines: 30, members: 25 */
        /* sum members: 1851, holes: 4, sum holes: 13 */
        /* last cacheline: 8 bytes */
After:
        /* size: 1856, cachelines: 29, members: 25 */
        /* sum members: 1851, holes: 1, sum holes: 5 */

struct _buffer {
        /* size: 80, cachelines: 2, members: 7 */
        /* sum members: 73, holes: 2, sum holes: 7 */
        /* last cacheline: 16 bytes */
After:
        /* size: 80, cachelines: 2, members: 7 */
        /* padding: 7 */
        /* last cacheline: 16 bytes */

struct _inode_table {
        /* size: 256, cachelines: 4, members: 26 */
        /* sum members: 233, holes: 5, sum holes: 20 */
        /* padding: 3 */
After:
        /* size: 240, cachelines: 4, members: 26 */
        /* padding: 7 */
        /* last cacheline: 48 bytes */

struct _fd {
        /* size: 120, cachelines: 2, members: 10 */
        /* sum members: 105, holes: 2, sum holes: 8 */
        /* padding: 7 */
        /* last cacheline: 56 bytes */
After:
        /* size: 112, cachelines: 2, members: 10 */
        /* padding: 7 */
        /* last cacheline: 48 bytes */

struct _client {
      /* size: 200, cachelines: 4, members: 14 */
        /* sum members: 192, holes: 2, sum holes: 8 */
        /* last cacheline: 8 bytes */
After:
       /* size: 192, cachelines: 3, members: 14 */

struct rpc_req {
        /* size: 496, cachelines: 8, members: 10 */
        /* sum members: 488, holes: 1, sum holes: 4 */
        /* padding: 4 */
        /* last cacheline: 48 bytes */
After:
        /* size: 488, cachelines: 8, members: 10 */
        /* last cacheline: 40 bytes */

struct syncenv {
        /* size: 16032, cachelines: 251, members: 12 */
        /* sum members: 16024, holes: 1, sum holes: 4 */
        /* padding: 4 */
        /* last cacheline: 32 bytes */
After:
        /* size: 16024, cachelines: 251, members: 12 */
        /* last cacheline: 24 bytes */

struct glusterd_brickinfo {
        /* size: 25968, cachelines: 406, members: 30 */
        /* sum members: 25953, holes: 2, sum holes: 8 */
        /* padding: 7 */
        /* last cacheline: 48 bytes */
After:
        /* size: 25960, cachelines: 406, members: 30 */
        /* sum members: 25953, holes: 1, sum holes: 4 */
        /* padding: 3 */
        /* last cacheline: 40 bytes */

struct ios_conf {
        /* size: 6792, cachelines: 107, members: 21 */
        /* sum members: 6769, holes: 4, sum holes: 19 */
        /* padding: 4 */
        /* last cacheline: 8 bytes */
After:
        /* size: 6776, cachelines: 106, members: 21 */
        /* sum members: 6769, holes: 2, sum holes: 7 */
        /* last cacheline: 56 bytes */

struct posix_aio_cb {
        /* size: 272, cachelines: 5, members: 9 */
        /* sum members: 264, holes: 2, sum holes: 8 */
        /* last cacheline: 16 bytes */
After:
        /* size: 264, cachelines: 5, members: 9 */
        /* last cacheline: 8 bytes */

Updates: #1000
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/glusterfs/circ-buff.h b/libglusterfs/src/glusterfs/circ-buff.h
index c51cf618f..54033abd6 100644
--- a/libglusterfs/src/glusterfs/circ-buff.h
+++ b/libglusterfs/src/glusterfs/circ-buff.h
@@ -28,7 +28,6 @@ struct _buffer {
     int used_len; /* indicates the amount of circular buffer used. */
     size_t size_buffer;
 
-
     circular_buffer_t **cb;
     void (*destroy_buffer_data)(void *data);
     pthread_mutex_t lock;
diff --git a/libglusterfs/src/glusterfs/glusterfs.h b/libglusterfs/src/glusterfs/glusterfs.h
index 686a9cb8b..7c9cff86f 100644
--- a/libglusterfs/src/glusterfs/glusterfs.h
+++ b/libglusterfs/src/glusterfs/glusterfs.h
@@ -583,7 +583,6 @@ struct _cmd_args {
     int fuse_flush_handle_interrupt;
     int fuse_auto_inval;
 
-
     uint32_t fuse_dev_eperm_ratelimit_ns;
 
     bool global_threading;
@@ -670,11 +669,11 @@ struct _glusterfs_ctx {
 
     pthread_t sigwaiter;
     char *cmdlinestr;
-    int graph_id;        /* Incremented per graph, value should
-                            indicate how many times the graph has
-                            got changed */
-    pid_t mnt_pid;       /* pid of the mount agent */
-    int process_mode;    /*mode in which process is runninng*/
+    int graph_id;     /* Incremented per graph, value should
+                         indicate how many times the graph has
+                         got changed */
+    pid_t mnt_pid;    /* pid of the mount agent */
+    int process_mode; /*mode in which process is runninng*/
     int mem_acct_enable;
     struct syncenv *env; /* The env pointer to the synctasks */
 
diff --git a/libglusterfs/src/glusterfs/inode.h b/libglusterfs/src/glusterfs/inode.h
index a392a78f0..0a88468e1 100644
--- a/libglusterfs/src/glusterfs/inode.h
+++ b/libglusterfs/src/glusterfs/inode.h
@@ -43,7 +43,7 @@ struct _inode_table {
     struct list_head *name_hash;  /* buckets for dentry hash table */
     struct list_head active; /* list of inodes currently active (in an fop) */
     uint32_t active_size;    /* count of inodes in active list */
-    uint32_t lru_limit;     /* maximum LRU cache size */
+    uint32_t lru_limit;      /* maximum LRU cache size */
     struct list_head lru;    /* list of inodes recently used.
                                 lru.next most recent */
     uint32_t lru_size;       /* count of inodes in lru list  */
@@ -55,8 +55,8 @@ struct _inode_table {
     struct mem_pool *fd_mem_pool; /* memory pool for fd_t */
     int ctxcount;                 /* number of slots in inode->ctx */
 
-    uint32_t root_id;    /* Save the xlator id at the time of inode table
-                            creation */
+    uint32_t root_id; /* Save the xlator id at the time of inode table
+                         creation */
     /* This is required for 'invalidation' when 'nlookup' would be used,
        specially in case of fuse-bridge */
     int32_t (*invalidator_fn)(xlator_t *, inode_t *);
diff --git a/libglusterfs/src/glusterfs/syncop.h b/libglusterfs/src/glusterfs/syncop.h
index 940fd3171..4c48cb563 100644
--- a/libglusterfs/src/glusterfs/syncop.h
+++ b/libglusterfs/src/glusterfs/syncop.h
@@ -146,7 +146,6 @@ struct syncenv {
                     so that no more synctasks are accepted*/
 
     size_t stacksize;
-
 };
 
 typedef enum { LOCK_NULL = 0, LOCK_TASK, LOCK_THREAD } lock_type_t;
```

mykaul commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/bugs/glusterfs/bug-866459.t

0 test(s) generated core 


6 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/bugs_core_multiplex-limit-issue-151.t
./tests/basic/afr/afr-read-hash-mode.t
./tests/bugs/glusterd/brick-order-check-add-brick.t
./tests/bugs/glusterd/bug-1696046.t
./tests/bugs/glusterfs/bug-866459.t
https://build.gluster.org/job/gh_centos7-regression/2322/

mykaul commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/bugs/shard/bug-1605056.t

0 test(s) generated core 


6 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/00-geo-rep/georep-basic-tarssh-ec.t
./tests/bugs/replicate/bug-1250170-fsync.t
./tests/bugs/shard/bug-1605056.t

2 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
https://build.gluster.org/job/gh_centos7-regression/2470/

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/glusterfs/glusterfs.h b/libglusterfs/src/glusterfs/glusterfs.h
index 408e83391..65812fcc4 100644
--- a/libglusterfs/src/glusterfs/glusterfs.h
+++ b/libglusterfs/src/glusterfs/glusterfs.h
@@ -63,7 +63,6 @@
 #define O_PATH 010000000 /* from asm-generic/fcntl.h */
 #endif
 
-
 #ifndef EBADFD
 /* Mac OS X does not have EBADFD */
 #define EBADFD EBADF
diff --git a/xlators/performance/write-behind/src/write-behind.c b/xlators/performance/write-behind/src/write-behind.c
index 19a0939cd..0512635e6 100644
--- a/xlators/performance/write-behind/src/write-behind.c
+++ b/xlators/performance/write-behind/src/write-behind.c
@@ -55,10 +55,10 @@ typedef struct wb_inode {
                                which arrive later (which overlap, etc.)
                                are issued only after their dependencies
                                in this list are "fulfilled".
-
+ 
                                Server acks for entries in this list
                                shrinks the window.
-
+ 
                                The sum total of all req->write_size
                                of entries in this list must be kept less
                                than the permitted window size.
@@ -92,14 +92,14 @@ typedef struct wb_inode {
                                     the current 'state' of liability. Every
                                     new addition to the liability list bumps
                                     the generation number.
-
+               
                                     a newly arrived request is only required
                                     to perform causal checks against the entries
                                     in the liability list which were present
                                     at the time of its addition. the generation
                                     number at the time of its addition is stored
                                     in the request and used during checks.
-
+               
                                     the liability list can grow while the request
                                     waits in the todo list waiting for its
                                     dependent operations to complete. however
```

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/glusterfs/glusterfs.h b/libglusterfs/src/glusterfs/glusterfs.h
index 408e83391..65812fcc4 100644
--- a/libglusterfs/src/glusterfs/glusterfs.h
+++ b/libglusterfs/src/glusterfs/glusterfs.h
@@ -63,7 +63,6 @@
 #define O_PATH 010000000 /* from asm-generic/fcntl.h */
 #endif
 
-
 #ifndef EBADFD
 /* Mac OS X does not have EBADFD */
 #define EBADFD EBADF
diff --git a/xlators/performance/write-behind/src/write-behind.c b/xlators/performance/write-behind/src/write-behind.c
index 19a0939cd..0512635e6 100644
--- a/xlators/performance/write-behind/src/write-behind.c
+++ b/xlators/performance/write-behind/src/write-behind.c
@@ -55,10 +55,10 @@ typedef struct wb_inode {
                                which arrive later (which overlap, etc.)
                                are issued only after their dependencies
                                in this list are "fulfilled".
-
+ 
                                Server acks for entries in this list
                                shrinks the window.
-
+ 
                                The sum total of all req->write_size
                                of entries in this list must be kept less
                                than the permitted window size.
@@ -92,14 +92,14 @@ typedef struct wb_inode {
                                     the current 'state' of liability. Every
                                     new addition to the liability list bumps
                                     the generation number.
-
+               
                                     a newly arrived request is only required
                                     to perform causal checks against the entries
                                     in the liability list which were present
                                     at the time of its addition. the generation
                                     number at the time of its addition is stored
                                     in the request and used during checks.
-
+               
                                     the liability list can grow while the request
                                     waits in the todo list waiting for its
                                     dependent operations to complete. however
```

mykaul commented:
/run regression

xhernandez reviewed (commented):
Looks fine, but it will need to be updated to fix the conflicts.

Commit:
Multiple files: reduce padding in structures

Examples:
struct gf_log_handle_
        /* size: 208, cachelines: 4, members: 22 */
        /* sum members: 194, holes: 2, sum holes: 8 */
        /* padding: 6 */
        /* last cacheline: 16 bytes */
After:
        /* size: 200, cachelines: 4, members: 22 */
        /* padding: 6 */
        /* last cacheline: 8 bytes */

struct clienttable {
        /* size: 72, cachelines: 2, members: 5 */
        /* sum members: 64, holes: 2, sum holes: 8 */
        /* last cacheline: 8 bytes */
After:
        /* size: 64, cachelines: 1, members: 5 */

struct gf_flock {
       /* size: 1056, cachelines: 17, members: 6 */
        /* sum members: 1052, holes: 1, sum holes: 4 */
        /* last cacheline: 32 bytes */
After:
        /* size: 1056, cachelines: 17, members: 6 */
        /* padding: 4 */
        /* last cacheline: 32 bytes */

struct _call_stack
        /* size: 1864, cachelines: 30, members: 25 */
        /* sum members: 1851, holes: 4, sum holes: 13 */
        /* last cacheline: 8 bytes */
After:
        /* size: 1856, cachelines: 29, members: 25 */
        /* sum members: 1851, holes: 1, sum holes: 5 */

struct _buffer {
        /* size: 80, cachelines: 2, members: 7 */
        /* sum members: 73, holes: 2, sum holes: 7 */
        /* last cacheline: 16 bytes */
After:
        /* size: 80, cachelines: 2, members: 7 */
        /* padding: 7 */
        /* last cacheline: 16 bytes */

struct _inode_table {
        /* size: 256, cachelines: 4, members: 26 */
        /* sum members: 233, holes: 5, sum holes: 20 */
        /* padding: 3 */
After:
        /* size: 240, cachelines: 4, members: 26 */
        /* padding: 7 */
        /* last cacheline: 48 bytes */

struct _fd {
        /* size: 120, cachelines: 2, members: 10 */
        /* sum members: 105, holes: 2, sum holes: 8 */
        /* padding: 7 */
        /* last cacheline: 56 bytes */
After:
        /* size: 112, cachelines: 2, members: 10 */
        /* padding: 7 */
        /* last cacheline: 48 bytes */

struct _client {
      /* size: 200, cachelines: 4, members: 14 */
        /* sum members: 192, holes: 2, sum holes: 8 */
        /* last cacheline: 8 bytes */
After:
       /* size: 192, cachelines: 3, members: 14 */

struct rpc_req {
        /* size: 496, cachelines: 8, members: 10 */
        /* sum members: 488, holes: 1, sum holes: 4 */
        /* padding: 4 */
        /* last cacheline: 48 bytes */
After:
        /* size: 488, cachelines: 8, members: 10 */
        /* last cacheline: 40 bytes */

struct syncenv {
        /* size: 16032, cachelines: 251, members: 12 */
        /* sum members: 16024, holes: 1, sum holes: 4 */
        /* padding: 4 */
        /* last cacheline: 32 bytes */
After:
        /* size: 16024, cachelines: 251, members: 12 */
        /* last cacheline: 24 bytes */

struct glusterd_brickinfo {
        /* size: 25968, cachelines: 406, members: 30 */
        /* sum members: 25953, holes: 2, sum holes: 8 */
        /* padding: 7 */
        /* last cacheline: 48 bytes */
After:
        /* size: 25960, cachelines: 406, members: 30 */
        /* sum members: 25953, holes: 1, sum holes: 4 */
        /* padding: 3 */
        /* last cacheline: 40 bytes */

struct ios_conf {
        /* size: 6792, cachelines: 107, members: 21 */
        /* sum members: 6769, holes: 4, sum holes: 19 */
        /* padding: 4 */
        /* last cacheline: 8 bytes */
After:
        /* size: 6776, cachelines: 106, members: 21 */
        /* sum members: 6769, holes: 2, sum holes: 7 */
        /* last cacheline: 56 bytes */

struct posix_aio_cb {
        /* size: 272, cachelines: 5, members: 9 */
        /* sum members: 264, holes: 2, sum holes: 8 */
        /* last cacheline: 16 bytes */
After:
        /* size: 264, cachelines: 5, members: 9 */
        /* last cacheline: 8 bytes */

Updates: #1000
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/glusterfs/glusterfs.h b/libglusterfs/src/glusterfs/glusterfs.h
index 408e83391..65812fcc4 100644
--- a/libglusterfs/src/glusterfs/glusterfs.h
+++ b/libglusterfs/src/glusterfs/glusterfs.h
@@ -63,7 +63,6 @@
 #define O_PATH 010000000 /* from asm-generic/fcntl.h */
 #endif
 
-
 #ifndef EBADFD
 /* Mac OS X does not have EBADFD */
 #define EBADFD EBADF
diff --git a/xlators/protocol/client/src/client.h b/xlators/protocol/client/src/client.h
index 880d3f8b1..bc5902e5b 100644
--- a/xlators/protocol/client/src/client.h
+++ b/xlators/protocol/client/src/client.h
@@ -95,8 +95,8 @@ typedef struct clnt_conf {
     rpc_clnt_prog_t *dump;
 
     int client_id;
-    int event_threads; /* # of event threads
-                        * configured */
+    int event_threads;        /* # of event threads
+                               * configured */
     uint64_t reopen_fd_count; /* Count of fds reopened after a
                                  connection is established */
     gf_lock_t rec_lock;
@@ -123,7 +123,7 @@ typedef struct clnt_conf {
                                   */
     gf_boolean_t filter_o_direct; /* if set, filter O_DIRECT from
                                      the flags list of open() */
-    gf_boolean_t send_gids; /* let the server resolve gids */
+    gf_boolean_t send_gids;       /* let the server resolve gids */
 
     gf_boolean_t destroy; /* if enabled implies fini was called
                            * on @this xlator instance */
@@ -136,7 +136,7 @@ typedef struct clnt_conf {
                                       * logged
                                       */
 
-    gf_boolean_t old_protocol;         /* used only for old-protocol testing */
+    gf_boolean_t old_protocol; /* used only for old-protocol testing */
     gf_boolean_t fini_completed;
     gf_boolean_t strict_locks; /* When set, doesn't reopen saved fds after
                                   reconnect if POSIX locks are held on them.
@@ -145,8 +145,8 @@ typedef struct clnt_conf {
                                   complaince as bricks cleanup any granted
                                   locks when a client disconnects.
                                */
-    gf_boolean_t connection_to_brick; /*True from attempt to connect to brick
-                                        till disconnection to brick*/
+    gf_boolean_t connection_to_brick;  /*True from attempt to connect to brick
+                                         till disconnection to brick*/
     pthread_cond_t fini_complete_cond; /* Used to wait till we finsh the fini
                                           compltely, ie client_fini_complete
                                           to return*/
```

mykaul commented:
/run regression
